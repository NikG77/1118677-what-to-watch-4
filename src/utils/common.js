import moment from 'moment';

export const formatForDateTime = (date) => moment(date).format(`YYYY-MM-DD`);

export const reformatDate = (date) => moment(date).format(`LL`);
// без использования moment
// export const reformatDate = (date) => new Intl.DateTimeFormat(`en-EN`, {year: `numeric`, month: `long`, day: `numeric`}).format(new Date(date));

export const formateDuration = (time) => {
  const hours = Math.trunc(time / 3600);
  const minutes = Math.trunc(time / 60 % 60);
  const seconds = Math.trunc(time % 60);

  return `${hours}:${minutes}:${seconds}`;
};

export const extend = (a, b) => {
  return Object.assign({}, a, b);
};
