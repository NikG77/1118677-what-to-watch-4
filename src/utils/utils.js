import moment from 'moment';
import Swal from "sweetalert2";

export const formatForDateTime = (date) => moment(date).format(`YYYY-MM-DD`);

export const reformatDate = (date) => moment(date).format(`LL`);
// без использования moment
// export const reformatDate = (date) => new Intl.DateTimeFormat(`en-EN`, {year: `numeric`, month: `long`, day: `numeric`}).format(new Date(date));

export const formateDuration = (time) => {
  const hours = Math.trunc(time / 3600);
  const minutes = Math.trunc(time / 60 % 60);
  const seconds = Math.trunc(time % 60);

  return `${hours}:${minutes}:${seconds}`;
};

export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const errorPopup = (err) => {
  if (!err.response && !err.request) {
    err.response.status = `Нет соединения`;
    err.response.data.error = `Проверьте соедение с интернетом`;
  } else if (!err.response && err.request) {
    err.response.status = `Oops... ${err.request.status}`;
    err.response.data.error = err.request.data.error;
  }
  return Swal.fire({
    icon: `error`,
    title: `Oops... ${err.response.status}`,
    text: err.response.data.error,
  });
};
