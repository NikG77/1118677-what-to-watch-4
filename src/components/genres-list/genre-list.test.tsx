import * as React from "react";
import * as renderer from "react-test-renderer";
import GenresList from "./genres-list";
import {noon} from "../../utils/utils";


const activeGenre: string = `All genre`;

it(`Should GenresList render correctly`, () => {
  const tree = renderer
    .create(<GenresList
      genresList={[activeGenre]}
      activeGenre={activeGenre}
      onGenreItemClick={noon}
    />
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
