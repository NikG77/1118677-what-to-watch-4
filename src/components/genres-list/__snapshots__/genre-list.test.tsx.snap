// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should GenresList render correctly 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item catalog__genres-item--active"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      All genre
    </a>
  </li>
</ul>
`;
